admin/test/test_release.py

```diff
@@ -661,3 +667,637 @@ def test_publish_non_release_fails(self):
+            cli_package: 'cli-package',
+            'clusterhq-flocker-node-0.3.3-0.dev.7.noarch.rpm': 'node-package',
+        }
+
+        self.update_repo(
+            aws=aws,
+            yum=FakeYum(),
+            rpm_directory=self.rpm_directory,
+            target_bucket=self.target_bucket,
+            target_key=self.target_key,
+            source_repo=self.create_fake_repository(files=repo_contents),
+            packages=self.packages,
+        )
+
+        expected_keys = existing_s3_keys.copy()
+        expected_keys.update({
+            os.path.join(self.target_key, package): repo_contents[package]
+            for package in repo_contents})
+
+        self.assertDictContainsSubset(
+            expected_keys,
+            aws.s3_buckets[self.target_bucket])
+
+    def test_repository_metadata_index_updated(self):
+        """
+        Calling :func:`update_repo` updates the repository metadata index.
+        """
+        index_path = os.path.join(self.target_key, 'repodata', 'repomd.xml')
+        existing_s3_keys = {
+            index_path: 'old_metadata_index',
+        }
+
+        aws = FakeAWS(
+            routing_rules={},
+            s3_buckets={
+                self.target_bucket: existing_s3_keys.copy(),
+            },
+        )
+
+        self.update_repo(
+            aws=aws,
+            yum=FakeYum(),
+            rpm_directory=self.rpm_directory,
+            target_bucket=self.target_bucket,
+            target_key=self.target_key,
+            source_repo=self.create_fake_repository(files={}),
+            packages=self.packages,
+        )
+
+        self.assertNotEqual(
+            existing_s3_keys[index_path],
+            aws.s3_buckets[self.target_bucket][index_path])
+
+    def test_existing_metadata_files_not_uploaded(self):
+        """
+        Calling :func:`update_repo` does not update repository metadata files
+        which are not the index.
+        """
+        index_path = os.path.join(self.target_key, 'repodata', 'repomd.xml')
+        existing_metadata_file = os.path.join(self.target_key, 'repodata',
+                                              'filelists.xml.gz')
+
+        existing_s3_keys = {
+            index_path: 'old_metadata_index',
+            existing_metadata_file: 'old_metadata_content',
+        }
+
+        aws = FakeAWS(
+            routing_rules={},
+            s3_buckets={
+                self.target_bucket: existing_s3_keys.copy(),
+            },
+        )
+
+        self.update_repo(
+            aws=aws,
+            yum=FakeYum(),
+            rpm_directory=self.rpm_directory,
+            target_bucket=self.target_bucket,
+            target_key=self.target_key,
+            source_repo=self.create_fake_repository(files={}),
+            packages=self.packages,
+        )
+
+        self.assertEqual(
+            aws.s3_buckets[self.target_bucket][existing_metadata_file],
+            existing_s3_keys[existing_metadata_file])
```

Comment:
This assertion doesn't make sense to me. Maybe it would be better if the fake `CreateRepository` actually included (fake) hashes in the names of the files, and this checked that the old ones are still there with the old data.